Yosemite/Yosemite/Stores/ProductStore.swift

```diff
@@ -229,11 +229,14 @@ private extension ProductStore {
                 if case NetworkError.notFound = error {
                     self.deleteStoredProduct(siteID: siteID, productID: productID)
                 }
-                onCompletion(nil, error)
+                onCompletion(.failure(error))
             case .success(let product):
                 self.upsertStoredProductsInBackground(readOnlyProducts: [product]) { [weak self] in
-                    let storageProduct = self?.storageManager.viewStorage.loadProduct(siteID: siteID, productID: productID)
-                    onCompletion(storageProduct?.toReadOnly(), nil)
+                    guard let storageProduct = self?.storageManager.viewStorage.loadProduct(siteID: siteID, productID: productID) else {
+                        onCompletion(.failure(ProductLoadError.notFoundInStorage))
+                        return
```

Comment:
nit: I think you merge this two into one line!
```suggestion
                        return onCompletion(.failure(ProductLoadError.notFoundInStorage))
```

Reply:
TIL'ed! updated in https://github.com/woocommerce/woocommerce-ios/pull/2939/commits/9fd4e2606316f8aeb751caab43ab81a79c314d0c